chapters/en/slides/module7/module7_02.Rmd

```diff
@@ -0,0 +1,342 @@
+    pandoc_args: "--markdown-headings=atx"
+---
+
+```{r setup, include=FALSE}
+source('../../../../slide-setup.R')
+# Remember to also manually update the YAML title above
+```
+
+type: slides
+
+
+# Advanced Selections
+
+Notes:
+In this chapter we will learn how to create new types of selections
+such as clicks and hovers.
+We will also see how to link selections between charts
+that do not stem from the same base chart,
+and how to add selections to geographical visualizations.
+
+---
+
+## Selections can be triggered on clicks instead of dragging with the mouse
+
+```{python}
+import altair as alt
+from vega_datasets import data
+
+cars = data.cars()
+click = alt.selection_multi()
+bars = (alt.Chart(cars).mark_bar().encode(
+    x='count()',
+    y='Origin',
+    color='Origin',
+    opacity=alt.condition(click, alt.value(0.9), alt.value(0.2)))
+.add_selection(click)).properties(width=300)
+
+bars
+```
+
+Notes:
+When creating charts such as barcharts and maps, 
+it is often more intuitive to be able to click
+the objects to select 
+rather than dragging with the mouse.
+
+In this slide we use `selection_multi`
+to create a selection based on what we click with the mouse.
+If we hold the shift key,
+we can select multiple bars
+(to disable this use `selection_single` instead).
+
+You can also see that instead of changing the color of the bar on selection,
+we here use `alt.condition` to change the opacity instead.
+Whether we change the color or opacity on selection is mostly an aesthetic choice.
+
+We assign the chart to a variable name because we will reuse it in the next slide
+and set the dimensions of the plot to fit on later slides 
+with more complex layouts.
+
+---
+
+## Combining charts with different selections
+
+```{python}
+brush = alt.selection_interval()
+points = (alt.Chart(cars).mark_circle().encode(
+    x='Horsepower',
+    y='Miles_per_Gallon',
+    color=alt.condition(brush, 'Origin', alt.value('lightgray')))
+.add_selection(brush)).properties(width=300, height=200)
+
+points & bars
+```
+
+Notes:
+To add two charts together,
+we can use what we learned previously about Altair's layout operators.
+To concatenate the charts vertically,
+we're here using the `&` operator.
+
+In contrast to the two scatterplots in an earlier slide,
+these charts are not created from the same base chart,
+so there selections are not linked.
+
+What if we want to link the selections
+so that when we select a region of origin in the bar chart
+the points in the scatter plot that are from the same origin would be highlighted?
+
+This would be useful to study observations from each Origin separately,
+so how can we perform this linkage in Altair?
+
+---
+
+## Linking selections from separate charts requires explicit columns to select on
+
+```{python}
+click = alt.selection_multi(fields=['Origin'])
+bars = bars.add_selection(click)
+points = points.encode(opacity=alt.condition(click, alt.value(0.9), alt.value(0.2)))
+
+points & bars
```

Comment:
Reusing the scatter plot and barplot code here, and overwriting the points and bar selection is a bit confusing and adds unnecessary cognitive load in my opinion. Let's just make a fresh scatter plot and bar chart from scratch each time please.

I know space might be an issue, but if you do horizontal concatenation, you might have enough space?